docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+
+    
+**_"registerTable"_** API takes the name of the table and data store endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for data. Each of these endpoints requires a different type of configurations but all of them demand data format to be defined. In this example data format is defined using _**MessageFormat**_ by specifying it at endpoint creation. As of now 
+**_CSVMessageFormat_** is only supported **_Messageformat_** for _KafkaEndpoint_ and _FileEndpoint_. **_CSVMessageFormat_** can be configured with schema string as follows:
+
+    {"separator":",","quoteChar":"\"","fields":[{"name":"RowTime","type":"Date","constraints":{"format":"dd/MM/yyyy hh:mm:ss Z"}},{"name":"id","type":"Integer"},{"name":"Product","type":"String"},{"name":"units","type":"Integer"}]}
+
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Currently, integration supports following data endpoints:
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
+* **FileEndpoint**
+: It needs to be configured with HDFS output folder, output file name and MessageFormat as in line 2 in above code 
+* **StreamEndpoint** 
+: This allows us to connect existing operator output or input to SQL query as a data source or sink respectively. StreamEndpoint needs immediate downstream operator's input port or immediate upstream operator's output port and field mapping for CSV data or POJO class. This will be explained in detail in next [example](http://192.168.1.31:8000/apis/calcite/#fustion-style-sql-application).
+
+
+### Using User Defined functions ( UDF ) in SQL query
+
+We can use our own scalar UDF, implemented in Java, in a SQL statement for data manipulation. But before the usage, we need to register the function with execution environment by using registerFunction API.
+
+    sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str")
+
+In above code, **_registerFunction_** takes UDF name to be used in SQL, JAVA class which implements **static** method and name of same method as parameters and returns instance of SQLExecEnvironment. 
+Used static method ‘apex_concat_str" takes two String objects as input parameters as a requirement from SQL query.
+
+    public static String apex_concat_str(String s1, String s2)
+
+    {
+
+     return s1 + s2;
+
+    }
+
+The scalar UDF "APEXCONCAT" can be used in SQL as follows whereas other standard SQL scalar functions such as FLOOR, SUBSTRING are provided by Apache Calcite. To read about functions and operators supported by Apache Calcite, click [here]("https://calcite.apache.org/docs/reference.html#operators-and-functions").
```

Comment:
Put etc after FLOOR, SUBSTRING etc... The sentence tells its limited to FLOOR and SUBSTRING only.